frontend/packages/console-app/src/components/nodes/NodeLogs.tsx

```diff
@@ -255,75 +257,108 @@ const NodeLogs: React.FC<NodeLogsProps> = ({ obj: node }) => {
-  const onChangeUnit = (value: string) => {
-    setUnit(value);
-    value === ''
-      ? removeQueryArgument(unitQueryArgument)
-      : setQueryArgument(unitQueryArgument, value);
-  };
-  const onChangeFilename = (event: React.ChangeEvent<HTMLInputElement>, newFilename: string) => {
-    event.preventDefault();
-    setFilenameOpen(false);
-    setLogFilename(newFilename);
-    setLoadingLog(true);
-    setQueryArgument(logQueryArgument, newFilename);
-    setLogURL(getLogURL(`/${newFilename}`));
-    fetchLog(getLogURL(`/${newFilename}`));
-    trimmedContent = '';
-  };
-  const onToggleFilename = () => setFilenameOpen(!isFilenameOpen);
+  const onChangePath = React.useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>, newAPI: string) => {
+      event.preventDefault();
+      setPathOpen(false);
+      setPath(newAPI);
+      setLogFilenames([]);
+      setLogFilename('');
+      setUnit('');
+      setQueryArgument(pathQueryArgument, newAPI);
+      removeQueryArgument(unitQueryArgument);
+      removeQueryArgument(logQueryArgument);
+      setLoadingFilenames(true);
+      setLoadingLog(true);
+      trimmedContent.current = '';
+    },
+    [pathQueryArgument, unitQueryArgument, logQueryArgument],
+  );
+  const onTogglePath = React.useCallback(() => setPathOpen(!isPathOpen), [isPathOpen]);
+  const onChangeUnit = React.useCallback(
+    (value: string) => {
+      setUnit(value);
+      value === ''
+        ? removeQueryArgument(unitQueryArgument)
+        : setQueryArgument(unitQueryArgument, value);
+    },
+    [unitQueryArgument],
+  );
+  const onChangeFilename = React.useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>, newFilename: string) => {
+      event.preventDefault();
+      setFilenameOpen(false);
+      setLogFilename(newFilename);
+      setLoadingLog(true);
+      setQueryArgument(logQueryArgument, newFilename);
+      setLogURL(getLogURL(`/${newFilename}`));
+      fetchLog(getLogURL(`/${newFilename}`));
+      trimmedContent.current = '';
+    },
+    [fetchLog, getLogURL],
+  );
+  const onToggleFilename = React.useCallback(() => setFilenameOpen(!isFilenameOpen), [
+    isFilenameOpen,
+  ]);
   const errorExists = error.length > 0;
   const logFilenamesExist = logFilenames.length > 0;
 
-  const logControls = (
-    <LogControls
-      onTogglePath={onTogglePath}
-      onChangePath={onChangePath}
-      path={path}
-      isPathOpen={isPathOpen}
-      pathItems={pathItems}
-      isJournal={isJournal}
-      onChangeUnit={onChangeUnit}
-      unit={unit}
-      isLoadingFilenames={isLoadingFilenames}
-      logFilenamesExist={logFilenamesExist}
-      onToggleFilename={onToggleFilename}
-      onChangeFilename={onChangeFilename}
-      logFilename={logFilename}
-      isFilenameOpen={isFilenameOpen}
-      logFilenames={logFilenames}
-      isWrapLines={isWrapLines}
-      setWrapLines={setWrapLines}
-    />
+  const logControls = React.useCallback(
```

Comment:
You want `React.useMemo`, and it should take `showSearch` in the dependency array.